src/execution_plan/ops/op_merge.c

```diff
@@ -74,9 +73,9 @@ static void _UpdateProperties(OpMerge *op, Record *records) {
 				if(t == REC_TYPE_NODE) _UpdateIndices(gc, (Node *)ge); // Update indices if necessary.
 			}
 		}
+		if(op->stats) op->stats->properties_set += update_count * record_count;
```

Comment:
Can this update move to after we unlock?

Reply:
unlock requires statistics updates in order to replicate